e2e/testcases/helm_sync_test.go

```diff
@@ -64,6 +64,18 @@ func TestPublicHelm(t *testing.T) {
 	if err := nt.Validate("my-ingress-nginx-controller", "ingress-nginx", &appsv1.Deployment{}); err != nil {
 		nt.T.Error(err)
 	}
+	// reset rs to sync from original git repo
+	nt.MustMergePatch(rs, fmt.Sprintf(`{"spec": {"sourceType": "%s", "helm": null, "git": {"dir": "acme", "branch": "main", "repo": "%s", "auth": "ssh","gcpServiceAccountEmail": "", "secretRef": {"name": "git-creds"}}}}`,
```

Comment:
Why do we need to reset the root sync? Can we patch it directly by setting `namespace` to empty?

Unnecessary source spec change will result in reconciler restart, which increases the running time.

Reply:
I tried to only change the namespace, but the WaitForRepoSyncs function does not work well with this condition. Since the RootSha1 and SyncDirectory are both same for these two syncs, it will not wait until the second sync finish.

Reply:
Can you confirm if that's an issue of the `WaitForRepoSyncs` function or the `helm-sync` container? I think it might because the image version is the same, so even if we update the `namespace` field, it won't trigger a re-sync.

And in this case, we should probably support resync.

Reply:
Both WaitForRepoSyncs and helm-sync container work fine. It turns out to be that I did not use the MustMergePatch() function correctly. Now I can patch root sync directly by setting namespace to empty.